metadata-ingestion/src/datahub/ingestion/graph/client.py

```diff
@@ -1103,3 +1251,17 @@ def close(self) -> None:
 def get_default_graph() -> DataHubGraph:
     (url, token) = get_url_and_token()
     return DataHubGraph(DatahubClientConfig(server=url, token=token))
+
+
+def _get_schema_info_from_graphql_response(schema: dict) -> "SchemaInfo":
+    return {
+        DatasetUrn.get_simple_field_path_from_v2_field_path(field["fieldPath"]): (
+            # The actual types are more of a "nice to have".
+            field["nativeDataType"]
+            or "str"
+        )
+        for field in schema["fields"]
+        # TODO: We can't generate lineage to columns nested within structs yet.
+        if "."
+        not in DatasetUrn.get_simple_field_path_from_v2_field_path(field["fieldPath"])
```

Comment:
this processing logic is pretty specific to sqlglot and probably belongs in sqlglot_lineage.py instead of here

feel free to modify / add methods to SchemaResolver e.g. `_process_raw_schema_info` or something

Reply:
moved this bit to `sqlglot_lineage`, added method `add_graphql_schema_metadata` in `SchemaResolver`